src/DynamoCore/Core/PathManager.cs

```diff
@@ -173,6 +174,23 @@ public bool ResolveLibraryPath(ref string library)
             return library != default(string);
         }
 
+        public bool ResolveDocumentPath(ref string document)
+        {
+            if (string.IsNullOrEmpty(document))
+            {
+                throw new ArgumentNullException();
+            }
+
```

Comment:
I think before the following lines, let's try to see if we can get the full path of the document first. Here's how:

```csharp
try
{
    document = Path.GetFullPath(document);
    if (File.Exists(document)) // "document" is already an absolute path.
        return true;

    // Restore "document" back to just its file name first...
    var document = Path.GetFileName(document);

    // Search alongside the main assembly location...
    var executingAssemblyPathName = ...;

    // Bla bla bla... (your existing code).

    // Finally...
    return File.Exists(document);
}
catch(Exception)
{
    return false;
}
```